src/gc/gc.d

```diff
@@ -1739,8 +1737,8 @@ struct Gcx
         // Return next item from free list
         bucket[bin] = (cast(List*)p).next;
         auto pool = (cast(List*)p).pool;
-        if (bits)
-            pool.setBits((p - pool.baseAddr) >> pool.shiftBy, bits);
+        pool.clrBits((p - pool.baseAddr) >> pool.shiftBy, ~bits);
+        pool.setBits((p - pool.baseAddr) >> pool.shiftBy, bits);
```

Comment:
Yeah, that's what I meant in the description: change the `setBits(biti, mask)` function to `setBits(biti, bits, mask)` that both sets and clears bits found in `mask`. It will still touch all bits in the case of malloc, though.